clar2wasm/src/standard/standard.wat

```diff
@@ -2522,6 +2536,293 @@
         (local.get $hi)
     )
 
+
+    (func $stdlib.sha512-buf (param $offset i32) (param $length i32) (param $offset-result i32)(result i32 i32)
+
```

Comment:
Could you please add a link to the original paper, or anything that can help understand the algorithm?